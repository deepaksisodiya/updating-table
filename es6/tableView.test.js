const tableRow = require('./TableView');

describe('TableView', () => {
  it('Call The Constructor and create table', () => {
    const div = document.createElement("div");
    const TableView = new tableRow({
      node: div,
      data: {
        gbpaud: {
          "name": "gbpaud",
          "bestBid": 1.8899720617188756,
          "bestAsk": 1.9550019816411313,
          "openBid": 1.8885006037024021,
          "openAsk": 1.932899396297598,
          "lastChangeAsk": 0.06332772101079831,
          "lastChangeBid": 0.08900450895206324
        }
      }
    });

    expect(div.innerHTML).toMatchSnapshot();
  });

  it('Sorted the given data by lastChangeBid', () => {
    const div = document.createElement("div");
    const TableView = new tableRow({
      node: div,
      data: {
        usdjpy: {
          "name": "usdjpy",
          "bestBid": 106.7297012204255,
          "bestAsk": 107.25199883791178,
          "openBid": 107.22827132623534,
          "openAsk": 109.78172867376465,
          "lastChangeAsk": -4.862314256927661,
          "lastChangeBid": 2
        },
        usdeur: {
          "name": "usdeur",
          "bestBid": 106.7297012204255,
          "bestAsk": 107.25199883791178,
          "openBid": 107.22827132623534,
          "openAsk": 109.78172867376465,
          "lastChangeAsk": -4.862314256927661,
          "lastChangeBid": -10
        },
        name: {
          "name": "euraud",
          "bestBid": 106.7297012204255,
          "bestAsk": 107.25199883791178,
          "openBid": 107.22827132623534,
          "openAsk": 109.78172867376465,
          "lastChangeAsk": -4.862314256927661,
          "lastChangeBid": 1
        },
      }
    });

    expect(div.innerHTML).toMatchSnapshot();
  });
});
