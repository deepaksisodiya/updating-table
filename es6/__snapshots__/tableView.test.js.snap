// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableView Call The Constructor and create table 1`] = `
"<table>
      <tbody><tr>
        <th>Name</th>
        <th>bestBid</th> 
        <th>bestAsk</th>
        <th>openBid</th>
        <th>openAsk</th>
        <th>lastChangeAsk</th>
        <th>lastChangeBid</th>
      </tr>
      <tr>
        <td>gbpaud</td>
        <td>1.8899720617188756</td> 
        <td>1.9550019816411313</td> 
        <td>1.8885006037024021</td> 
        <td>1.932899396297598</td> 
        <td>0.06332772101079831</td> 
        <td>0.08900450895206324</td>
      </tr>
    </tbody></table>"
`;
